common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java

```diff
@@ -47,35 +47,45 @@ public void onTrackedQuestUpdate(TrackedQuestUpdateEvent event) {
     }
 
     @SubscribeEvent
-    public void onRenderLevelLast(RenderLevelLastEvent event) {
+    public void onRenderLevelLast(RenderTileLevelLastEvent event) {
         if (CompassModel.getCompassLocation().isEmpty()) return;
 
         PoseStack poseStack = event.getPoseStack();
-        MultiBufferSource.BufferSource bufferSource =
-                MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
+        MultiBufferSource.BufferSource bufferSource = MultiBufferSource.immediate(new BufferBuilder(256));
 
-        Player player = McUtils.player();
+        Vec3 camera = event.getCamera().getPosition();
+        Location location = CompassModel.getCompassLocation().get();
 
-        poseStack.pushPose();
-        float partials = event.getPartialTick();
-        Camera camera = McUtils.mc().gameRenderer.getMainCamera();
-        poseStack.translate(-camera.getPosition().x, -camera.getPosition().y, -camera.getPosition().z);
+        double dx = location.x - camera.x;
+        double dy = location.y - camera.y;
+        double dz = location.z - camera.z;
 
-        Location location = CompassModel.getCompassLocation().get();
-        poseStack.translate(location.x, location.y, location.z);
+        double distance = MathUtils.magnitude(dx, dz);
+        int maxDistance = (McUtils.mc().options.renderDistance - 1) * 16;
```

Comment:
this might be a stupid question but why renderDistance - 1? Why not renderdistance?

Reply:
Also oops